### ISSUE TITLE
[Flaky test] [sig-windows] ci-kubernetes-e2e-aks-engine-azure-master-windows-2004

### ISSUE THREAD

**Issue by jsturtevant** [kubernetes/kubernetes#93590]
**Which jobs are flaking**:
ci-kubernetes-e2e-aks-engine-azure-master-windows-2004

**Which test(s) are flaking**:
the whole test run occasionally is bad

causing error:

```
_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e.go:76
Jul 30 01:15:21.574: Unexpected error:
    <*errors.errorString | 0xc0026f78f0>: {
        s: "there are currently no ready, schedulable nodes in the cluster",
    }
    there are currently no ready, schedulable nodes in the cluster
occurred
_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e.go:267
```

**Testgrid link**:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-aks-engine-azure-master-windows-2004/1288631402773876737

**Reason for failure**:
There is a race with the windows nodes becoming available. In the build log:

```
2020/07/30 01:15:16 e2e.go:474: Cluster nodes:
k8s-master-15399475-0   Ready   master   2m49s   v1.20.0-alpha.0.416+0c642b6ef01071
```

happens right before the windows nodes came online:

```
I0730 01:15:23.486979    5056 kubelet_node_status.go:70] Attempting to register node 1539k8s000
I0730 01:15:23.500740    5056 kubelet_node_status.go:73] Successfully registered node 1539k8s000
```

**Anything else we need to know**:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

/sig windows
/cc @chewong

**Comment by jsturtevant** [comment]
This appears to be the case for what happened with https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-aks-engine-azure-1-18-windows/1288134789127213057 only the node was not ready:

```
 Jul 28 16:46:26.140: INFO: Condition Ready of node 1807k8s000 is false, but Node is tainted by NodeController with [{node.kubernetes.io/unreachable  NoSchedule 2020-07-28 16:04:53 +0000 UTC} {node.kubernetes.io/unreachable  NoExecute 2020-07-28 16:04:58 +0000 UTC} {node.cloudprovider.kubernetes.io/shutdown  NoSchedule <nil>}]. Failure
Jul 28 16:46:26.140: INFO: Unschedulable nodes:
Jul 28 16:46:26.140: INFO: -> 1807k8s000 Ready=false Network=false Taints=[{node.kubernetes.io/unreachable  NoSchedule 2020-07-28 16:04:53 +0000 UTC} {node.kubernetes.io/unreachable  NoExecute 2020-07-28 16:04:58 +0000 UTC} {node.cloudprovider.kubernetes.io/shutdown  NoSchedule <nil>}] NonblockingTaints:node-role.kubernetes.io/master
Jul 28 16:46:26.140: INFO: ================================
Jul 28 16:46:26.140: FAIL: Unexpected error:
    <*errors.errorString | 0xc0000a7fe0>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred 
```

```
2020/07/28 16:16:21 e2e.go:474: Cluster nodes:
1807k8s000              NotReady   agent    20m   v1.18.7-rc.0.8+ec73e191f47b79
1807k8s001              Ready      agent    20m   v1.18.7-rc.0.8+ec73e191f47b79
1807k8s010              Ready      agent    20m   v1.18.7-rc.0.8+ec73e191f47b79
k8s-master-18078817-0   Ready      master   20m   v1.18.7-rc.0.8+ec73e191f47b79
```

**Comment by chewong** [comment]
Looks like 1807k8s000 just crashed. Can't even ssh into it:
```
Collecting logs for vm 1807k8s000
Testing connection to host 1807k8s000.
Warning: Permanently added 'kubetest-d290930a-d0e7-11ea-a7e2-92d2b23710df.eastus2.cloudapp.azure.com,13.77.127.169' (ECDSA) to the list of known hosts.
channel 0: open failed: connect failed: Connection refused
stdio forwarding failed
ssh_exchange_identification: Connection closed by remote host
```

**Comment by jsturtevant** [comment]
The reported error has been resolved by kubernetes-sigs/windows-testing#187, Haven't see the other issue were the windows node crashes.

/close
